tests/unit/mixins/run-test.js

```diff
@@ -124,6 +124,86 @@ test('runTask tasks can be canceled', function(assert) {
   }, 10);
 });
 
+test('scheduleTask invokes async tasks', function(assert) {
+  assert.expect(2);
+
+  let subject = this.subject();
+  let done = assert.async();
+  let hasRun = false;
+
+  run(() => {
+    subject.scheduleTask('actions', () => {
+      hasRun = true;
+      assert.ok(true, 'callback was called');
+      done();
+    });
+
+    assert.notOk(hasRun, 'callback should not have run yet');
```

Comment:
I'm suggesting an assertion for `assert.ok(hasRun` after the runloop, as that would show when the task was supposed to run (during the runloop queue flush). As written, this test doesn't assert timing beyond a) not immediately and b) eventually. I think you want to also assert c) after the runloop `scheduleTask` is called in. At the least. You could go further and assert what queue it runs in, but that seems ok to pass on to me.